source/fundamentals/serialization/poco.txt

```diff
@@ -261,6 +262,166 @@ The following code sample maps the ``Identifier`` property to the
    The ``_id`` field mapping logic described in this section only applies to the
    root document and does not apply to nested documents.
 
+Specify an ID Generator
+```````````````````````
+
+Every document in a MongoDB collection must have a unique ID. When you serialize an object
+to a collection, if its ID property contains the default
+value for its data type (usually ``null``), the {+driver-short+} doesn't serialize the
+default value. Instead, the driver generates a unique ID value and assigns it to the
```

Comment:
The C# driver ***TRIES*** to generate a unique Id value but can onlly do so in cases where it knows what Id generator to use.